Add Groestlcoin (GRS)

HashEngineering opened the pull request:
This PR will add support for Groestlcoin addresses.  Groestlcoin uses a new hash for the checksum of the addresses and that is imported from npm instead of including the file directly with the hash function as with Blake.

Since according to my comment on issue #10 this library only supports bech32 addresses for bitcoin, the tests for bech32 addresses have been commented out.

ajoslin committed:
Add vertcoin

ajoslin committed:
Covnert to hex

martin-key committed:
Added bitcoin gold

Commit:
added Garlicoin

cernael committed:
Add new LTC P2SH prefix

ognus committed:
Merge pull request #6 from keyhash/master

Update jsSHA library from version 1.6.0 to 2.3.1.

ognus committed:
Merge pull request #5 from dylmye/master

Add Garlicoin

ognus committed:
Dropped bower, fixed travis CI config. Updated README.

ognus committed:
Merge pull request #9 from ognus/cleanup

Dropped bower, fixed travis CI config. Updated README.

ognus committed:
Added missing test and build

ognus committed:
Merge pull request #10 from ognus/ajoslin-patch-1

Ajoslin patch 1

ognus committed:
Merge branch 'patch-1' of git://github.com/cernael/wallet-address-validator into cernael-patch-1

ognus committed:
Merge branch 'cernael-patch-1'

ognus committed:
Added missing test and build, udpated README

ognus committed:
Merge branch 'StarbuckBG-master'

ognus committed:
chore(release): 0.1.2

R3VoLuT1OneR committed:
Extended currencies list with Decred

R3VoLuT1OneR committed:
Browserify package

ognus committed:
Merge pull request #11 from R3VoLuT1OneR/master

Extended currencies list with Decred

ognus committed:
Added Ethereum and Digibyte currencies

ognus committed:
Blake256 fix for browser. Dropped Grunt, added Karma test runner and moved to Browserify.

ognus committed:
Travis CI workaround for Headless Chrome, https://github.com/travis-ci/travis-ci/issues/8836

ognus committed:
Merge pull request #13 from ognus/blake

Blake256 fix and cleanup

ognus committed:
Merge pull request #14 from ognus/ethereum

Added Ethereum and Digibyte currencies

ognus committed:
chore(release): 0.1.3

dpominov committed:
Add ripple validator

dpominov committed:
add test for ripple

dpominov committed:
Add Ripple to README.md

ognus committed:
Merge pull request #18 from dpominov/add_ripple

Add ripple vlidation

ognus committed:
Added .editorconfig and fixed whitespaces

ognus committed:
Small code refactor

ognus committed:
Added Dash and listed coins alphabetically in README.md

ognus committed:
Merge pull request #19 from ognus/dash

Dash

ognus committed:
chore(release): 0.1.4

zeppelin committed:
Add Bitcoin Cash

mcuelenaere committed:
Add support for validating NEO addresses

mcuelenaere committed:
Update README about NEO addition

ognus committed:
Merge pull request #20 from zeppelin/add-bcash

Add Bitcoin Cash

ognus committed:
Merge branch 'mcuelenaere-neo-support'

ognus committed:
chore(release): 0.1.5

ognus committed:
Code clean and refactor

ognus committed:
Merge pull request #25 from ognus/fun_bind

Code clean and refactor

ognus committed:
Added Qtum validation

ognus committed:
Merge pull request #26 from ognus/qtum

Qtum validation

ognus committed:
chore(release): 0.1.6

hellcatz committed:
Update currencies.js

Add votecoin, zencash, zclassic, zcash, snowgem, hush, bitcoinprivate, bitcoinz, komodo

hellcatz committed:
Update wallet_address_validator.js

Small optimization

iddoberger committed:
add neogas

iddoberger committed:
change test value

iddoberger committed:
add gas to keywords in package.json

iddoberger committed:
add neogas to dist

hellcatz committed:
Update wallet_address_validator.js

Add tests for VOTE, BTCZ, ZCl, HUSH, ZEC, BTCP, SNG, ZEN, KMD

hellcatz committed:
Add files via upload

Add equihash coins to browser files

hellcatz committed:
Update wallet_address_validator.js

Added invalid tests for equihash coins

ognus committed:
Merge pull request #29 from hellcatz/hellcatz-equihash

Add some equihash coins

ognus committed:
Merge branch 'iddoberger-master'

ognus committed:
Added new coins to Readme

ognus committed:
chore(release): 0.1.7

iobaixas committed:
chore(): add failing spec for P2SH testnet litecoin address

Currently the library only supports legacy testnet P2SH addresses

iobaixas committed:
feat(litecoin): adds support for new testnet P2SH addresses

ognus committed:
Merge pull request #31 from iobaixas/patch-1

Add support for new P2SH testnet litecoin addresses

ognus committed:
chore(dist): rebuilt the bundle file

mrqc committed:
regex ripple improved

dominique-blockchain committed:
feat: segwit support for bitcoin

ognus committed:
Merge pull request #35 from deefactorial/master

feat: segwit support for bitcoin

ognus committed:
refactored segwit support

PavelVoronov committed:
added new currency bkx-bankex, tests attached

ognus committed:
Merge pull request #38 from PavelVoronov/master

added new currency bkx-bankex, tests attached

ognus committed:
chore(release): 0.2.1

mcuelenaere committed:
Add Blake2B support to crypto utils (taken from https://github.com/emilbayes/blake2b)

mcuelenaere committed:
Add support for validating NANO/XRB addresses

mcuelenaere committed:
Update README about NANO addition

maximumgame committed:
Add monero/xmr validation

christsim committed:
added support for cardano

christsim committed:
Revert "regex ripple improved"

This reverts commit 4cc61759a96dce06ace1495d4ec464d95c104604.

christsim committed:
added xmr from a fork

christsim committed:
added coins.  mainly eth tokens

christsim committed:
added expanse

christsim committed:
clean up.

christsim committed:
fixed ada browser compatibility

christsim committed:
updated readme.md to reflect current coins

christsim committed:
publicshing to npm

christsim committed:
fixed trailing comma

christsim committed:
updated package.json

christsim committed:
this commit adds waves support

mcuelenaere committed:
Regenerate dist

mcuelenaere committed:
bump deps

christsim committed:
Merge https://github.com/getdelta/wallet-address-validator into merge_nano

christsim committed:
added rlc, salt, and storj

christsim committed:
added more erc20 coins (Viberate, Odyssey, Polymath, Storm)

christsim committed:
removed unneccesary toArrayBuffer method

christsim committed:
removed duplicate base58 decoding package

eddiewang committed:
add sc and xsc

eddiewang committed:
merge

eddiewang committed:
add lbry

eddiewang committed:
add loki

christsim committed:
Merge https://github.com/LuxorLabs/wallet-address-validator

christsim committed:
v0.3.5

christsim committed:
v0.3.5

fixed qtum testnet

updated to v0.3.6

adds regtest address checking for bitcoin

Adds Address Validation For Tron (TRX)

Adds Wallet Address Validation For XEM

Adds New Validators To Readme

Adds Wallet Address Validation For LSK

Removes lisk library

Removes unnecessary libraries

Removes crypto-js library

Clean up

Adds BSV validator

Update bch to use bsv validator

Changes from bsv_validator to bch_validator

Adds extra validators (#3)

* Adds Address Validation For Tron (TRX)

* Adds Wallet Address Validation For XEM

* Adds New Validators To Readme

* Adds Wallet Address Validation For LSK

* Removes lisk library

* Removes unnecessary libraries

* Removes crypto-js library

* Clean up

updated version

Adds bundle and minify versions

Adds check prefix for bch/bsv (bitcoincash)

Updates README

Adds XLM address validator

fixed vertcoin validator for '3' addresses

Comment Out Currencies Generator

Adds XTZ (Tezos) address validator

Updates README to include XTZ

updated dependencies

added yarn.lock

updated to 0.3.10

added multiple erc20 tokens

updated some dependencies
renamed some coin names

added get currencies and find currencies

updated travis to include latest node and remove old node versions

updated README.md

cherry-picked eos validator

Cherry picked and added EOS

updated version

Adds XTZ address validator

Adds XTZ address validator (#13)

updated version after merging tezos

adds support for monero sub address

christsim committed:
fixed qtum testnet

christsim committed:
updated to v0.3.6

christsim committed:
adds regtest address checking for bitcoin

sasiedu committed:
Adds extra validators (#3)

* Adds Address Validation For Tron (TRX)

* Adds Wallet Address Validation For XEM

* Adds New Validators To Readme

* Adds Wallet Address Validation For LSK

* Removes lisk library

* Removes unnecessary libraries

* Removes crypto-js library

* Clean up

christsim committed:
updated version

sasiedu committed:
Adds BSV (#4)

* Adds Address Validation For Tron (TRX)

* Adds Wallet Address Validation For XEM

* Adds New Validators To Readme

* Adds Wallet Address Validation For LSK

* Removes lisk library

* Removes unnecessary libraries

* Removes crypto-js library

* Clean up

* Adds BSV validator

* Update bch to use bsv validator

* Changes from bsv_validator to bch_validator

* Adds bundle and minify versions

* Adds check prefix for bch/bsv (bitcoincash)

sasiedu committed:
Updates README to include BCH (#5)

* Adds Address Validation For Tron (TRX)

* Adds Wallet Address Validation For XEM

* Adds New Validators To Readme

* Adds Wallet Address Validation For LSK

* Removes lisk library

* Removes unnecessary libraries

* Removes crypto-js library

* Clean up

* Adds BSV validator

* Update bch to use bsv validator

* Changes from bsv_validator to bch_validator

* Adds bundle and minify versions

* Adds check prefix for bch/bsv (bitcoincash)

* Updates README

christsim committed:
fixed vertcoin validator for '3' addresses

sasiedu committed:
Adds XLM address validator (#6)

* Adds Address Validation For Tron (TRX)

* Adds Wallet Address Validation For XEM

* Adds New Validators To Readme

* Adds Wallet Address Validation For LSK

* Removes lisk library

* Removes unnecessary libraries

* Removes crypto-js library

* Clean up

* Adds BSV validator

* Update bch to use bsv validator

* Changes from bsv_validator to bch_validator

* Adds bundle and minify versions

* Adds check prefix for bch/bsv (bitcoincash)

* Updates README

* Adds XLM address validator

* Comment Out Currencies Generator

christsim committed:
updated dependencies

christsim committed:
added yarn.lock

christsim committed:
Merge branch 'master' of github.com:christsim/multicoin-address-validator

christsim committed:
updated to 0.3.10

GusGold committed:
cherry-picked eos validator

christsim committed:
added multiple erc20 tokens

updated some dependencies
renamed some coin names

christsim committed:
added get currencies and find currencies

christsim committed:
updated travis to include latest node and remove old node versions

christsim committed:
updated README.md

christsim committed:
Cherry picked and added EOS

christsim committed:
updated version

sasiedu committed:
Adds XTZ address validator

sasiedu committed:
Adds XTZ address validator

sasiedu committed:
Adds XTZ address validator (#13)

christsim committed:
updated version after merging tezos

HashEngineering committed:
Add Groestlcoin (GRS)

HashEngineering commented:
If this PR needs to be rebased I am happy to do that if you don't use Squash and Merge.

gruve-p commented:
@christsim Let us know if this PR is ok to get merged or something else needs to be fixed also

sasiedu committed:
Merge remote-tracking branch 'upstream/master' into chris_master

# Conflicts:
#	dist/wallet-address-validator.min.js
#	package.json
#	test/wallet_address_validator.js

sasiedu committed:
adds support for monero sub address

Adds XTZ address validator

rebase to upstream master

builds bundle

merges upstream master

sasiedu committed:
Merge remote-tracking branch 'origin/chris_master' into chris_master

sasiedu committed:
rebase to upstream master

sasiedu committed:
squash commits

sasiedu committed:
Merge remote-tracking branch 'origin/chris_master'

# Conflicts:
#	dist/wallet-address-validator.js
#	dist/wallet-address-validator.min.js
#	src/crypto/utils.js
#	src/nem_validator.js
#	test/wallet_address_validator.js

sasiedu committed:
merge upstream master

christsim committed:
added support for monero sub addresses,
updated depencies.

christsim committed:
merged lodash.isequal fix

christsim committed:
merged monero stagenet

gruve-p commented:
@christsim Fixed conflicts and ready for review. This has been implemented and tested on Trezor: https://github.com/trezor/trezor-address-validator/commit/b213f7a37495be5fb1a1ca0db2095e9809f2ccab

christsim committed:
updating yarn before running travis

christsim committed:
fixing travis build added install command

christsim committed:
removed node 8 from travis as it is out of lts maintenance.

gruve-p commented:
@christsim Fixed conflicts again and ready for review.

christsim committed:
Added VET and StormX

sasiedu committed:
Adds validation of usdt address for both erc20 and omni

sasiedu committed:
Merge branch 'master' into sasiedu-master

# Conflicts:
#	dist/wallet-address-validator.js
#	dist/wallet-address-validator.min.js
#	src/monero_validator.js
#	test/wallet_address_validator.js

sasiedu committed:
Merges upstream master

sasiedu committed:
Updates validate to change networkType to object

sasiedu committed:
Adds validation of both omni and erc20 addresses for Tether (#21)

christsim committed:
updated version

rraina committed:
Add LTC Bech32 Support (#22)

* Added bech32 support for litecoin

* Added dist files

christsim committed:
updated to version 0.4.7 after litecoin segwit pr merge.

christsim committed:
added cardano shelly support

gruve-p commented:
@christsim Let us know if you have a chance to review

sasiedu committed:
Merge branch 'master' into sasiedu-master

# Conflicts:
#	dist/wallet-address-validator.js
#	dist/wallet-address-validator.min.js
#	package.json
#	src/bitcoin_validator.js
#	src/monero_validator.js
#	test/wallet_address_validator.js
#	yarn.lock

sasiedu committed:
Adds segwit check for digibyte address

sasiedu committed:
Adds segwit check for digibyte address

sasiedu committed:
Adds segwit check for digibyte address

sasiedu committed:
Adds segwit support for digibyte (#30)

christsim committed:
Merged and fixed Digibyte segwit address validation

christsim committed:
Merged and fixed Digibyte segwit address validation
Added extra guard if segwit field is not present on currencies.

christsim committed:
added xtp
updated dependencies

christsim committed:
added support for new golem erc20 GLM
rename old golem to Golem (GNT)

sasiedu committed:
adds more erc20 validators

sasiedu committed:
Merge branch 'master' into sasiedu-master

# Conflicts:
#	dist/wallet-address-validator.js
#	dist/wallet-address-validator.min.js
#	package.json
#	src/currencies.js
#	test/wallet_address_validator.js

sasiedu committed:
Fixes buffer error and jssha error

sasiedu committed:
Adds more erc20 and fixes minor issues (#38)

* Adds XTZ address validator

* Adds XTZ address validator

* rebase to upstream master

* v0.3.5

fixed qtum testnet

updated to v0.3.6

adds regtest address checking for bitcoin

Adds Address Validation For Tron (TRX)

Adds Wallet Address Validation For XEM

Adds New Validators To Readme

Adds Wallet Address Validation For LSK

Removes lisk library

Removes unnecessary libraries

Removes crypto-js library

Clean up

Adds BSV validator

Update bch to use bsv validator

Changes from bsv_validator to bch_validator

Adds extra validators (#3)

* Adds Address Validation For Tron (TRX)

* Adds Wallet Address Validation For XEM

* Adds New Validators To Readme

* Adds Wallet Address Validation For LSK

* Removes lisk library

* Removes unnecessary libraries

* Removes crypto-js library

* Clean up

updated version

Adds bundle and minify versions

Adds check prefix for bch/bsv (bitcoincash)

Updates README

Adds XLM address validator

fixed vertcoin validator for '3' addresses

Comment Out Currencies Generator

Adds XTZ (Tezos) address validator

Updates README to include XTZ

updated dependencies

added yarn.lock

updated to 0.3.10

added multiple erc20 tokens

updated some dependencies
renamed some coin names

added get currencies and find currencies

updated travis to include latest node and remove old node versions

updated README.md

cherry-picked eos validator

Cherry picked and added EOS

updated version

Adds XTZ address validator

Adds XTZ address validator (#13)

updated version after merging tezos

adds support for monero sub address

* adds support for monero sub address

Adds XTZ address validator

rebase to upstream master

builds bundle

merges upstream master

* squash commits

* merge upstream master

* Adds validation of usdt address for both erc20 and omni

* Merges upstream master

* Updates validate to change networkType to object

* Adds segwit check for digibyte address

* Adds segwit check for digibyte address

* Adds segwit check for digibyte address

* adds more erc20 validators

* Fixes buffer error and jssha error

Co-authored-by: Chris Tsimogiannis <chris@valr.com>

christsim committed:
Merge branch 'master' of github.com:christsim/multicoin-address-validator

christsim committed:
updated version

christsim committed:
downgraded jsha

christsim committed:
downgraded jsha

Commit:
EOS addresses can include a decimal . (#37)

Examples: 

https://bloks.io/account/big.one
https://bloks.io/account/bp.dfs
https://bloks.io/account/hoo.com

gamalielhere committed:
Change GameCredits validation to ETH (#32)

* Taking out GameCredits

Please correct me if I'm wrong but it seems like GameCredits have moved to ERC20 https://medium.com/@gamecredits/game-credits-manual-redemption-process-for-july-1st-snapshot-fa13170e2ab3

* Move GameCredits to ETHValidator

Co-authored-by: christsim <christsim@gmail.com>

christsim committed:
GameCredits is not an ERC20 address
EOS supports a '.' in it the address

christsim committed:
Added algo

christsim committed:
added palkadot

gruve-p commented:
@christsim rebased on upstream to fix conflicts

christsim committed:
changed CRO to use new mainnet format
fixed stellar validator
updated bech32 and segwit address files

christsim committed:
Downgraded minify to support node 12 for now.

yashutanna committed:
Add Uniswap, Aave and Matic (#48)

* Add Uniswap, Aave and Matic

* add mana

Co-authored-by: yashu <yashu@valr.com>

christsim committed:
added aave, mana, matic, uni

gruve-p commented:
@christsim rebased

christsim committed:
added support for solana

christsim committed:
added support for solana - added missing files

christsim committed:
fix solana min length

christsim committed:
fix solana min length

Commit:
Merge branch 'master' of https://github.com/christsim/multicoin-address-validator into chris_master

Commit:
Merge branch 'chris_master'

# Conflicts:
#	README.md
#	dist/wallet-address-validator.js
#	dist/wallet-address-validator.min.js
#	package.json
#	src/crypto/utils.js
#	src/currencies.js

Commit:
fixes to support node v12

sasiedu committed:
Adds support for node 12 (#58)

* fixes to support node v12

christsim committed:
updated version, support for node 12

Commit:
merge confliect

Commit:
Merge branch 'christsim-master'

Commit:
adds bnb coin

Commit:
comments out log

christsim committed:
merged pr - adds bnb

imoark committed:
Fixed typo Error on Polkadot validator (#40)

christsim committed:
merged pr - fixed dot typo

gruve-p commented:
@christsim rebased again

gruve-p commented:
@christsim Is this PR unwanted?
